fastlane_core/lib/fastlane_core/tag_version.rb

```diff
@@ -0,0 +1,24 @@
+require "rubygems"
+
+module FastlaneCore
+  # Utility class to construct a Gem::Version from a tag.
+  # Accepts vX.Y.Z and X.Y.Z.
+  class TagVersion < Gem::Version
+    class << self
+      def correct?(tag)
+        superclass.correct?(version_number_from_tag(tag))
+      end
+
+      # Gem::Version.new barfs on things like "v0.1.0", which is the style
+      # generated by the rake release task. Just strip off any initial v
+      # to generate a Gem::Version from a tag.
+      def version_number_from_tag(tag)
+        tag.sub(/^v/, "")
```

Comment:
If the only things that work are versions of the form `1.0` or `1.2.3.4.5` it might be a good idea to strip away everything else. Something like `tag.sub(/\D*((?:\d\.?)+)\D*/, '\1')` might more accurately extract only the version number independently of what else is in that line

Reply:
RubyGems allows things like 1.3.0-beta.1, and 1.3.0-beta.2 > 1.3.0-beta.1.

```
[jdee@Jimmy-Dees-MacBookPro ~]$ irb
2.4.2 :001 > v1 = Gem::Version.new "1.3.0-beta.1"
 => #<Gem::Version "1.3.0.pre.beta.1"> 
2.4.2 :002 > v2 = Gem::Version.new "1.3.0-beta.2"
 => #<Gem::Version "1.3.0.pre.beta.2"> 
2.4.2 :003 > v2 <=> v1
 => 1 
```

I'm afraid that doing finer parsing here will be brittle and essentially rebuild bits of RubyGems. This method exists because RubyGems does not directly accept tags generated by the rake release task (from the Bundler, I believe).

In fact, I have a feeling that the reason this originally involved custom parsing might be because someone didn't want to be limited to whatever is supported by RubyGems (or didn't think of using RubyGems for non-Ruby version checking). After all lots of repos that use Fastlane are not Ruby repos and may have somewhat different versioning systems. But the option is either use something standard like RubyGems or, as @KrauseFx said somewhere, reinvent the wheel.

If a more general wheel ever did become necessary, this TagVersion class at least gives us a place to reinvent it.